File: leetcode/125-Valid-Palindrome/ValidPalindrome_001.py
class Solution:
    # @param {string} s
    # @return {boolean}
    def isPalindrome(self, s):
        p = 0
        num = len(s)
        q = num - 1
        
        while p < q:
            if s[p].isalnum() and s[q].isalnum():
                if s[p].lower() != s[q].lower():
                    return False
                else:
                    p = p + 1
                    q = q - 1
            else:
                while p <= q and s[p].isalnum() == False:
                    p = p + 1
                
                while p <= q and s[q].isalnum() == False:
                    q = q - 1
                
        return True

File: leetcode/125-Valid-Palindrome/ValidPalindrome_001.java
public class Solution {
    public boolean isPalindrome(String s) {
        int i = 0, j = s.length() - 1;
        
        while (i < j) {
            char chari = s.charAt(i);
            char charj = s.charAt(j);
            
            if (Character.isLetterOrDigit(chari)) {
                if (Character.isLetterOrDigit(charj)) {
                    char chi = Character.toLowerCase(chari);
                    char chj = Character.toLowerCase(charj);
                    if (chi != chj) {
                        return false;
                    } else {
                        i++;
                        j--;
                    }
                } else {
                    j--;
                }
            } else {
                i++;
            }
        }
        
        return true;
    }
}
